polaris-service/src/test/java/org/apache/polaris/service/task/TaskTestUtils.java

```diff
@@ -62,10 +69,32 @@ static ManifestFile manifestFile(
     return writer.toManifestFile();
   }
 
-  static void writeTableMetadata(FileIO fileIO, String metadataFile, Snapshot... snapshots)
+  static TableMetadata writeTableMetadata(FileIO fileIO, String metadataFile, Snapshot... snapshots)
       throws IOException {
-    TableMetadata.Builder tmBuidler =
-        TableMetadata.buildFromEmpty()
+    return writeTableMetadata(fileIO, metadataFile, null, null, null, snapshots);
+  }
+
+  static TableMetadata writeTableMetadata(FileIO fileIO, String metadataFile, List<StatisticsFile> statisticsFiles, Snapshot... snapshots)
+          throws IOException {
+    return writeTableMetadata(fileIO, metadataFile, null, null, statisticsFiles, snapshots);
+  }
+
+  static TableMetadata writeTableMetadata(
```

Comment:
Extend this API to write metadata with previous metadata and statistics file for each snapshot